### DISCUSSION TITLE
❗ GroupAvatar

### DISCUSSION THREAD

**Discussion by LaurentDario** [gsoft-inc/sg-orbit#1126] [New components Ideas]
## Description

_Initially discussed in #694_
Avatar works great for a user, with its placeholder using the initials of the user's Firstname and Lastname, it could be used also for a Group with a more complex name. This includes names with more than 2 capitalized word (for example _Name, Name and Name_) or a simple name with and `and` in the middle (for example _Something and Sons_), or some other funky names with dashes in the middle (for example _Management - People_). In all of these cases, the 2 letters Initials of the Avatar component are wrong (_NN_ for the first example, _Sa_ for the second and _M-_ for the last).

`GroupAvatar` could allow for a more complex filtering of the name to still output the correct name as aria-label while outputting the correct Initials as placeholder.

**Comment by fraincs** [comment]
How does Microsoft deals with that?

**Comment by LaurentDario** [comment]
Not sure what the magic between the Microsoft logic is but it looks like it goes for initials, sometimes figuring out how to do composed names (with dashes) but if it gets too complicated, they go for just the first initial of the name.

**Comment by fraincs** [comment]
Before moving on a clear spec of the Microsoft logic would be necessary, do we have ways to figure this out? Can any fullstack dev helps here?

**Comment by fraincs** [comment]
@alexasselin008 any thoughts on that issue?

**Comment by alexasselin008** [comment]
Can we show some examples of where this would be used in our applications? 

I believe the Avatar component was taken from fluid ui : https://developer.microsoft.com/en-us/fluentui#/controls/web/persona

They have a logic on how the generate user placeholder icons. As for the groups/teams, I think every groups have an image. If there is no user uploaded image, there is an image with initials that is returned. So would this component be used as a placeholder while we fetch the image? 

For instance, in our group list, here is an example of a group avatar with no user's uploaded image: 
![image](https://user-images.githubusercontent.com/38871812/162806022-de4d812e-0b71-4221-8088-51d83485357d.png)

**Comment by LaurentDario** [comment]
I think the Group aspect of the ticket is misleading, a complex name can also be found as a user name.
I'm not sure it is a big deal tho, but the avatar will, for example, use the "M" from "Mrs" or "Miss" as an initial:
![image](https://user-images.githubusercontent.com/234418/162959595-5d9134cc-1d79-4329-b97c-2991ed881ef3.png)
I don't think this needs to be addressed if we're ok with the way it is currently handled. It could be the consumer's job to handle that logic through the name passed to the Avatar component.

**Comment by alexasselin008** [comment]
The persona component in Fluent has a slightly different behavior: https://developer.microsoft.com/en-us/fluentui#/controls/web/persona

![image](https://user-images.githubusercontent.com/38871812/162972927-e0505919-887c-4d9b-94f6-fc6f9561f12d.png)

But the initials are not handled either.

**Comment by fraincs** [comment]
@alexasselin008 in that case we should not support initials either?

**Comment by patricklafrance** [comment]
Is this component still in play? If so, I believe it should be moved to Github Discussion until a formal proposal is available.
